twisted/internet/base.py

```diff
@@ -601,18 +602,33 @@ def sigInt(self, *args):
         """Handle a SIGINT interrupt.
         """
         log.msg("Received SIGINT, shutting down.")
+        try:
+            import signal
```

Comment:
`twisted.internet.posixbase` seems mostly concerned with ICP and not necessary signal handling.  (Other than SIGCHLD which is a special case).

`twisted.internet.base` was the only code that I saw that is actively intercepting SIGTERM and SIGINT and doing anything special with them.  That is the source of the bug - catching these signals, which the OS expects to cause a non-zero exit status.  Thus I piggybacked that code to ensure that after shutdown the appropriate exit status code is returned.

If the exit handling code should go somewhere else, it is not obvious to me where, or how it would fit into `twisted.internet.posixbase`  Please advise.